### ISSUE TITLE
Give original filename to phpcs

### ISSUE THREAD

**Issue by choma** [flycheck/flycheck#921]
As explained [here](https://github.com/squizlabs/PHP_CodeSniffer/issues/934#issuecomment-201022854), using file path instead of STDIN allows phpcs to infer which rules should be excluded.

**Comment by lunaryorn** [comment]
Please summarise the discussion in the issue you've referenced as far as it's relevant to Flycheck and elaborate why we need to drop standard input for phpcs.  Ideally, try to produce a minimal example where phpcs behaves different as expected with Flycheck.

**Comment by choma** [comment]
There are phpcs rules that *depends* on the file location.
As an example take this excerpt from CakePHP code sniffer ([taked from here](https://github.com/cakephp/cakephp-codesniffer/blob/master/CakePHP/ruleset.xml#L102)):
```xml
<rule ref="Squiz.NamingConventions.ValidFunctionName">
  <exclude-pattern>*/src/*</exclude-pattern>
  <exclude-pattern>*/tests/*</exclude-pattern>
</rule>
```
it tells to phpcs to exclude that rule when the path of the file being checked match the patterns.
So, if you are checking files under `src/` (which are almost all in a cakephp project and most other frameworks) and you call phpcs with file path it won't apply that rule, and you'll see less errors.
If you call phpcs with STDIN it can't know where the file is, so it applies _all_ the rules, resulting in an incorrect list of errors.
I hope it to be understandable, my english is really bad.

**Comment by lunaryorn** [comment]
@choma Other syntax checkers (e.g. rubocop, eslint, etc.) solve this by accepting the name of the file as argument.  Is that no option for phpcs?

**Comment by choma** [comment]
I copy my comment on flycheck issue, so you can find it easily:

> As I understand from [this](https://github.com/squizlabs/PHP_CodeSniffer/issues/733) issue, and what I can understand from [this](https://github.com/squizlabs/PHP_CodeSniffer/blob/0abb1dfc00b9b0fad870505224c588b6c3f9a7a1/CodeSniffer/File.php#L459) code, you have to add `phpcs_input_file: [file path]` as the first line of content.
> I've just tested it and it works great!

**Comment by lunaryorn** [comment]
@choma phpcs 2.6 will provide an option which allows us to set the path, see https://github.com/squizlabs/PHP_CodeSniffer/issues/934#issuecomment-203161356

When phpcs 2.6 is released will use this option to fix this issue.

**Comment by lunaryorn** [comment]
phpcs 2.6 is out now, I'll update the syntax checker definition soon.  Note that this will implicitly drop support for any older phpcs release.

**Comment by lunaryorn** [comment]
@choma I've updated phpcs accordingly, the issue should be fixed now.

Thanks for reporting it, and for your help with finding a proper solution! 👍
